backend/app/services/chat/task_default_knowledge_bases.py

```
@@ -44,41 +56,340 @@ def _build_task_knowledge_base_ref(
+
+def _get_ghosts_for_bots(
+    db: Session,
+    team: Kind,
+    bots: list[Kind],
+) -> dict[tuple[str, str], Kind]:
+    """Batch load Ghosts referenced by Bot specs."""
+    ghost_refs: set[tuple[str, str]] = set()
+    for bot in bots:
+        try:
+            bot_crd = Bot.model_validate(bot.json)
+        except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)